src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachine/Config/NewAzureRmVMRunCommandConfigCommand.cs

```diff
@@ -0,0 +1,76 @@
+// 
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// 
+
+// Warning: This code was generated by a tool.
+// 
+// Changes to this file may cause incorrect behavior and will be lost if the
+// code is regenerated.
+
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Management.Compute.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet("New", "AzureRmVMRunCommandConfig", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSRunCommandInput))]
+    public partial class NewAzureRmVMRunCommandConfigCommand : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true)]
+        public string CommandId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true)]
+        public string[] Script { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true)]
+        public RunCommandInputParameter[] Parameter { get; set; }
```

Comment:
I think, in normal case, a user can do New-AzureRmVMRunCommandConfig without -Parameter and then pipe the object to Add-AzureRmVMRunCommandParameter to add a parameter object.  But, it is also possible a user can get an array of RunCommandInputParameter object using New-Object.